applications/portal/backend/api/migrations/0003_auto_20230512_1548.py

```diff
@@ -0,0 +1,59 @@
+import os
+import shutil
+
+from django.conf import settings
+from django.db import migrations
+
+from api.constants import POPULATIONS_DATA
+from api.models import PopulationStatus
+
+HERE = os.path.dirname(os.path.realpath(__file__))
+COLORS = ["#0000FF", "#FF0000", "#00FF00", "#FFA500"]
+
+
+def create_residential_populations(apps, schema_editor):
+    Population = apps.get_model('api', 'Population')
+    directory = os.path.join(HERE, '../../data/residential_populations')
+
+    for index, pop_dir in enumerate(os.listdir(directory)):
+        pop_dir_path = os.path.join(directory, pop_dir)
+
+        # Look for the CSV file in the directory
+        csv_file = next((f for f in os.listdir(pop_dir_path) if f.endswith('.csv')), None)
+        if not csv_file:
+            continue  # If no CSV file is found, skip this directory
+
+        name, _ = os.path.splitext(csv_file)
+        population = Population(
+            name=name,
+            color=COLORS[index % len(COLORS)],
+            opacity=1.0,
+            is_fiducial=False,
+            status=PopulationStatus.RUNNING
+        )
+        population.save()
+
+        # Copy the entire contents of each population's local directory to the storage path
+        storage_path = os.path.join(settings.PERSISTENT_ROOT, POPULATIONS_DATA, str(population.id))
```

Comment:
I had to use this here instead of population.storage_path because of django's use of [historical models](https://docs.djangoproject.com/en/2.2/topics/migrations/#historical-models)

Reply:
@afonsobspinto you could move the path construct to a "helper" or "service" function and then call the same from both places to make it more maintanable